src/java/boa/compiler/visitors/CodeGeneratingVisitor.java

```diff
@@ -326,13 +327,19 @@ public void visit(final TupleType n) {
 					fields.add("f" + fieldCount);
 				}
 				fieldCount++;
-				types.add(c.getType().type.toBoxedJavaType());
+				BoaType type = c.getType().type;
+				Boolean isProto = false;
```

Comment:
these 4 lines can be simplified into: `protos.add(type instanceof BoaProtoTuple);`